routes/cube_routes.js

```diff
@@ -1339,6 +1342,82 @@ async function bulkuploadCSV(req, res, cards, cube) {
   }
 }
 
+async function bulkuploadCSV(req, res, cards, cube) {
+  let changelog = '';
+  let { newCards, newMaybe, missing } = await CSVtoCards(cards);
+  newCards.forEach((card) => (changelog += addCardHtml(carddb.cardFromId(card.cardID))));
+  cube.cards.push(...newCards);
+  cube.maybe.push(...newMaybe);
+  await generateBlogpost(req, res, cube, changelog, newCards.concat(newMaybe), missing);
```

Comment:
this looks like suspicious chaining to me. it might be a better pattern to return the last 4 parameters and then have those passed into generateBlogpost